File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/c_solutions/965-univalued-binary-tree.c
#include <stdbool.h>

bool isUnivalTree(struct TreeNode* root) {
    if (root == NULL) {
        return true;
    }

    if (root->left != NULL) {
        if (root->left->val != root->val) {
            return false;
        }
        if (!isUnivalTree(root->left)) {
            return false;
        }
    }

    if (root->right != NULL) {
        if (root->right->val != root->val) {
            return false;
        }
        if (!isUnivalTree(root->right)) {
            return false;
        }
    }

    return true;
}

File: leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/965-univalued-binary-tree.rs
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn is_unival_tree(root: Option<Rc<RefCell<TreeNode>>>) -> bool {
        if let Some(node) = root {
            let node = node.borrow();
            let val = node.val;
            Self::helper(&node.left, val) && Self::helper(&node.right, val)
        } else {
            true
        }
    }

    fn helper(root: &Option<Rc<RefCell<TreeNode>>>, val: i32) -> bool {
        if let Some(node) = root {
            let node = node.borrow();
            node.val == val && Self::helper(&node.left, val) && Self::helper(&node.right, val)
        } else {
            true
        }
    }
}